### PR TITLE
like query fix

### PR DESCRIPTION
**Pull request by burakack**
## Motivation and Context
Now Devs can't query any non string fields

#96 close

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: src/Services/QueryService.ts [hunk 1]
```diff
@@ -502,6 +502,13 @@ class QueryService {
     if (this.hasSpecialStructure(where.field, structure)) {
       where.field = where.field.replace(structure, "");
       where.condition = condition;
+      where.model.columns.forEach(column=>{
```
Comment (ozziest) [inline src/Services/QueryService.ts:505]: We should check it only for `$like` and `$notLike` operators

**Review by ozziest — CHANGES_REQUESTED** [review]
We need integration tests.

PS: 
image

File: src/Services/QueryService.ts [hunk 2]
```diff
@@ -449,6 +449,13 @@ class QueryService {
       where.condition === ConditionTypes.LIKE ||
       where.condition === ConditionTypes["NOT LIKE"]
     ) {
+      where.model.columns.forEach((column) => {
+        if (column.name == where.field && column.data_type !== "varchar") {
+          throw new ApiError(
+            `Query field need to be varchar. Unacceptable query field: ${where.field}`
```
Comment (ozziest) [inline src/Services/QueryService.ts:455]:
```suggestion
            `You can not use the LIKE/NOT LIKE operator on non-varchar fields: ${where.field}`
```

**Review by ozziest — CHANGES_REQUESTED** [review]
@burakack , that's great. 👏 We are so close to merging the PR. Can you add one integration test, please? 

- Try to use LIKE operator on a non-varchar field, and test the error message is correct.

**Comment by ozziest** [comment]
@saracalihan is there anything that I can help?

**Comment by ozziest** [comment]
@burakack do you have any time to keep working on it? 😊

**Comment by burakack** [comment]
> @burakack do you have any time to keep working on it? 😊

Unfortunately, I can't even find time to drink water. Working life is very tiring. 😞 
I wish I had free time I can't wait to contribute the project

**Pull request closed without merging**